website/events/api/v2/admin/serializers/event_registration.py

```diff
@@ -0,0 +1,32 @@
+from rest_framework import serializers
+
+from events.models import EventRegistration
+from members.api.v2.serializers.member import MemberSerializer
+from members.models import Member
+
+
+class EventRegistrationSerializer(serializers.ModelSerializer):
+    """Serializer for event registrations."""
+
+    class Meta:
+        model = EventRegistration
+        fields = (
+            "pk",
+            "present",
+            "queue_position",
+            "date",
+            "date_cancelled",
+            "payment",
+            "member",
+            "name",
+        )
+
+    def to_internal_value(self, data):
+        self.fields["member"] = serializers.PrimaryKeyRelatedField(
+            queryset=Member.objects.all()
+        )
+        return super().to_internal_value(data)
+
+    def to_representation(self, instance):
+        self.fields["member"] = MemberSerializer(detailed=False)
```

Comment:
Going to `localhost:8000/api/v2/admin/events/11/registrations/7/` shows me the ability to `PUT` to a member profile, even if I do not have permissions for this. Maybe the MemberSerializer should have `readonly=True`?

Reply:
Idem for the organizer, btw

Reply:
That's only because those views assume that the output of a serializer is exactly the same as the input of a serializer. Which I changed in this case, since it accepts a different member id and not a full object. Don't think I can do anything to prevent that. And marking the field as read only could probably make it impossible to edit the field at all.

Reply:
Looks like it still works when I make them read only :)